# Exercise 1:

It is somewhat interesting that (5 * 4 * 3 * 2 * 1) perfectly divides (10 * 9 * 8 * 7 * 6) - there's no remainder.

If we only wanted exactly four heads as opposed to five, the equivalent calculation would be (10 * 9 * 8 * 7) / (4 * 3 * 2 * 1).

Does that evenly divide too? What is the formula in general?

Does it always come out as a positive whole number?

In [2]:
(10 * 9 * 8 * 7 * 6) / (5 * 4 * 3 * 2 * 1)

252.0

In [3]:
(10 * 9 * 8 * 7) / (4 * 3 * 2 * 1) 

210.0

In [3]:
import math
math.comb(10,4)

210

The calc in the question evenly divides with no remainder.

I researched mathematical formulas online and found one closest to the one given above called "Combinations and permutations". The combination formula used factorials to find the number of ways (combinations) of r things can be selected out of a sample of n.

If the order of the set is important, we would use the permutation formula, but in this case, it does not matter the order of heads selected out of 10 coin tosses, it is only important that we select exactly 4 heads thus the combination formula is used. The combinations formula is also called Binomial Coefficient (Pierce, 2021).

The formula is as follows;
 
n!/r!(n-r)!

For n ≥ r ≥ 0

where
n = total number of choices/coin tosses
r = number of choices we want/ number of heads or tails wanted

Given that;
- The trials are independent.
- There are only two possible outcomes per trial.

10!/4!(10-4)!
= (10x9x8x7x6x5x4x3x2x1)/(4x3x2x1)(6x5x4x3x2x1)
= 210
There are 210 ways to flip 10 coins to get exactly 4 heads.


It will always come out as a positive whole number as we are finding the number of ways of picking r out of n possibilities. n must be divisable by r, r is a subset of n, by definition the result cannot be negative and cannot be a fraction thus must be a whole number. I cannot get a negative/decimial result in getting 4 heads whilst flipping 10 coins.

# Exercise 2:

Note that there are the same number of ways to get 4 tails as there to get 4 heads. Explain why this is.


There are the same number of ways to select 4 tails out of 10 tosses as there are 4 heads as there are only two outcomes per trial.
If we use the above formula to find the number of ways 4 tails will be selected out of 10 tosses, r = 4 and n = 10, the answer will be 210, or using python math.comb method found on w3 schools as illustrated below.

In [4]:
# Import math Library
import math

# Initialize the number of items (coin toss trials) to choose from
n = 10

# Initialize the number of possibilities (exactly 4 tails) to choose
k = 4

# Print total number of possible combinations
print (math.comb(n, k))

210


# References

Pierce, Rod,  2021, 'Combinations and Permutations', Math Is Fun, Available at: <http://www.mathsisfun.com/combinatorics/combinations-permutations.html>. [Accessed 20 Oct 2022]

Pierce, Rod,  2022, 'The Binomial Distribution', Math Is Fun, Available at: <http://www.mathsisfun.com/data/binomial-distribution.html>. [Accessed 22 Oct 2022]

w3schools.com (no date) Python math.comb() Method, Python math.comb() method. Available at: https://www.w3schools.com/python/ref_math_comb.asp (Accessed: November 22, 2022). 